lib/sha256.c

```diff
@@ -491,15 +502,22 @@ static int my_sha256_final(unsigned char *out,
  * output [in/out] - The output buffer.
  * input  [in]     - The input data.
  * length [in]     - The input length.
+ *
+ * Returns CURLE_OK on success.
  */
-void Curl_sha256it(unsigned char *output, const unsigned char *input,
+CURLcode Curl_sha256it(unsigned char *output, const unsigned char *input,
                    const size_t length)
 {
+  CURLcode ret;
   my_sha256_ctx ctx;
 
-  my_sha256_init(&ctx);
+  ret = my_sha256_init(&ctx);
+  if(ret != CURLE_OK)
+    return ret;
+
   my_sha256_update(&ctx, input, curlx_uztoui(length));
   my_sha256_final(output, &ctx);
+  return ret;
 }
 
 
```

Comment:
I can't comment on line 527, but the `HMAC_hinit_func` signature (and the MD5 init function) would have to be updated too to return `CURLcode`. Unfortunately the `CURLX_FUNCTION_CAST` meant for the parameter hides that. Looks good to me otherwise!

Reply:
Sorry, I forgot to update it though you reminded me. I will do it after work...